### ISSUE TITLE
login_timeout not applying to asyncssh.connect()

### ISSUE THREAD

**Issue by jgauth** [ronf/asyncssh#415]
Python version: 3.8.9
asyncssh version: 2.7.2

I'm calling asyncssh.connect() with login_timeout=30, but it is still only timing out after the default value of 2 minutes.

Call looks like:
```
async with asyncssh.connect(host, username='user', password='pass', known_hosts=None, login_timeout=30) as conn:
```

Debug logs:
```
DEBUG:asyncio:Using selector: EpollSelector
INFO:asyncssh:Opening SSH connection to redacted_hostname, port 22
DEBUG:asyncio:Get address info redacted_hostname:22, type=<SocketKind.SOCK_STREAM: 1>
DEBUG:asyncio:Getting address info redacted_hostname:22, type=<SocketKind.SOCK_STREAM: 1> took 2.615ms: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('redacted_ip', 22))]

< hangs here for 2 minutes >

DEBUG:asyncio:Close <_UnixSelectorEventLoop running=False closed=False debug=True>
```
Let me know if there's anything else I could provide.
Thanks!

**Comment by ronf** [comment]
Hi John,

Thanks for the suggestion on the doc update. I think that's a good idea. Given the multiple questions I've had about this, I decided to take this a step further, though, and add a new connect_timeout connection option which is basically shorthand for calling asyncio.wait_for(). So, you can now do something like:

```python
async with asyncssh.connect(host, connect_timeout=30) as conn:
```
instead of
```python
async with asyncio.wait_for(asyncssh.connect(host), timeout=30)) as conn:
```
I've also now converted the "ConnectTimeout" client config file option to use this new capability, making it better match the behavior of OpenSSH. The config file support also means you can set different connect timeouts based on host or other matching criteria in the config file.

This new option applies any time an outbound connection is being made, even if that connection is used to set up a "reverse" SSH server connection, or even to set up a tunnel which is then used to open an SSH listener. It can also be used when connecting to an SSH server to retrieve host keys from it.

This change is available as commit b2cadab in the "develop" branch if you'd like to give it a try, and I'll be including this in the next release.

**Comment by jgauth** [comment]
Looks great, thanks a lot!

**Comment by ronf** [comment]
My pleasure! This new functionality is now released in AsyncSSH 2.8.0.
